### PR TITLE
fix(platform-browser): update template binding regex to match any string

### LINKED ISSUE #14423: ngFor: change detection crashes when array contains character code 0x2029
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

I've written a font viewer and stumbled over this problem: changing an array which is iterated via ngFor causes an exception when the array contains the character code 0x2029 (dec 8233). The errortext is "Cannot read property '1' of null" (Chrome) or "existingBindings is null" (Firefox)

The exception does not occur if AOT compiled.

```
TypeError: Cannot read property '1' of null
    at DomRenderer.setBindingDebugInfo (http://localhost:4200/vendor.bundle.js:33670:78)
    at DebugDomRenderer.setBindingDebugInfo (http://localhost:4200/vendor.bundle.js:61041:24)
    at setBindingDebugInfo (http://localhost:4200/vendor.bundle.js:22908:18)
    at http://localhost:4200/vendor.bundle.js:22893:9
    at Array.forEach (native)
    at setBindingDebugInfoForChanges (http://localhost:4200/vendor.bundle.js:22892:26)
    at Wrapper_NgFor.ngDoCheck (/CommonModule/NgFor/wrapper.ngfactory.js:50:7)
    at CompiledTemplate.proxyViewClass.View_AppComponent0.detectChangesInternal (/AppModule/AppComponent/component.ngfactory.js:149:20)
    at CompiledTemplate.proxyViewClass.AppView.detectChanges (http://localhost:4200/vendor.bundle.js:62650:14)
    at CompiledTemplate.proxyViewClass.DebugAppView.detectChanges (http://localhost:4200/vendor.bundle.js:62845:44)
    at CompiledTemplate.proxyViewClass.AppView.internalDetectChanges (http://localhost:4200/vendor.bundle.js:62635:18)
    at CompiledTemplate.proxyViewClass.View_AppComponent_Host0.detectChangesInternal (/AppModule/AppComponent/host.ngfactory.js:29:19)
    at CompiledTemplate.proxyViewClass.AppView.detectChanges (http://localhost:4200/vendor.bundle.js:62650:14)
    at CompiledTemplate.proxyViewClass.DebugAppView.detectChanges (http://localhost:4200/vendor.bundle.js:62845:44)
    at ViewRef_.detectChanges (http://localhost:4200/vendor.bundle.js:46648:20)
```

Simplified plnkr: http://plnkr.co/edit/xHUQjp4Mdr7gDSKb8Sly?p=preview

**Please tell us about your environment:**

Tested in Linux (Ubuntu 16.01) and Windows 7

* **Angular version:** 2.0.X

Tested with Angular 2.4.7 and 4.0.0-beta.7

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

Tested with Chrome 56 and Firefox 51
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

Tested with TypeScript 2.1.5

### PR DESCRIPTION
**Pull request by DzmitryShylovich**
Closes #14423

### CODE REVIEW

**Review by vicb — CHANGES_REQUESTED** [review]
The server renderer needs the same fix and you should also add tests. Thanks.

**Comment by DzmitryShylovich** [comment]
@vicb where can I find test specs for debug bindings? Didn't find anything.

File: modules/@angular/platform-browser/test/dom/dom_renderer_spec.ts [hunk 1]
```diff
@@ -13,24 +13,41 @@ import {BrowserModule} from '@angular/platform-browser';
...
 
 export function main() {
   describe('DomRenderer', () => {
 
-    beforeEach(() => TestBed.configureTestingModule({imports: [BrowserModule, TestModule]}));
+    describe('setBindingDebugInfo', () => {
+      describe('template binding regex', () => {
+        it('should match the string containing Unicode char with code 8233', () => {
+          // https://github.com/angular/angular/issues/14423
+          const expression = `{"ng-reflect-ng-if": "${String.fromCharCode(8233)}"}`;
```
Comment (vicb) [inline modules/@angular/platform-browser/test/dom/dom_renderer_spec.ts]: I'm not really sure if those added tests are really helpful... ie do we use that regexp at all ?

Integration tests would be much better. Could you take a stab at adding them ?

**Comment by DzmitryShylovich** [comment]
fix after 4.0 when api will be stable

**Pull request closed without merging**